scripts/setup.sh

```
@@ -0,0 +1,245 @@
+    else
+        echo "Creating virtual environment at $VENV_DIR..."
+        "$PYTHON" -m venv "$VENV_DIR"
+        echo "✓ .venv created"
+    fi
+    PYTHON_BIN="$VENV_DIR/bin/python"
+fi
+
+echo ""
+echo "Installing project dependencies..."
+"$PYTHON_BIN" -m pip install --quiet --upgrade pip
+"$PYTHON_BIN" -m pip install --quiet -e "$ROOT_DIR[dev]"
```

Shellcheck (0.11.0)

[error] Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).

(SC1087)